spec/sequel/plugins/enum_values_spec.rb

```diff
@@ -227,6 +227,150 @@
 						-> { expect { subject }.to raise_error(NoMethodError) }
 				end
 			end
+
+			shared_context('with filter_methods environment') do
+				subject(:enum_filter) { item_class.public_send(enum_value) }
+
+				def filter_sql(column, value)
+					"SELECT * FROM \"items\" WHERE (\"#{column}\" = '#{value}')"
+				end
+			end
+
+			shared_context('with filter_methods option') do |value|
+				before do
+					item_class.plugin :enum_values, filter_methods: value
+				end
+			end
+
+			describe 'filter_methods default option' do
+				before { item_class.plugin :enum_values }
+
+				include_context('with filter_methods environment')
+
+				type_enum_values.each do |filter_method_name|
+					describe filter_method_name do
+						let(:enum_value) { "type_#{filter_method_name}" }
+
+						it { expect { enum_filter }.to raise_error(NoMethodError) }
+					end
+				end
+
+				status_enum_values.each do |filter_method_name|
+					describe filter_method_name do
+						let(:enum_value) { "status_#{filter_method_name}" }
+
+						it { expect { enum_filter }.to raise_error(NoMethodError) }
+					end
+				end
+			end
+
+			describe 'filter_methods true option' do
+				include_context('with filter_methods environment')
+				include_context('with filter_methods option', true)
+
+				type_enum_values.each do |filter_method_name|
+					describe filter_method_name do
+						let(:enum_value) { "type_#{filter_method_name}" }
+
+						it { expect(enum_filter).to be_kind_of(Sequel::Dataset) }
+
+						it do
+							expect(enum_filter.sql).to(
```

Comment:
You can use `subject { enum_filter.sql }` and then `it { is_expected.to ... }`. `describe 'enum_filter.sql'` is optional for multiple test cases.

Reply:
Done